### ISSUE TITLE
Disk offload fails with bfloat16 weights

### ISSUE THREAD

**Issue by noamwies** [huggingface/accelerate#454]
### System Info

```Shell
- `Accelerate` version: 0.10.0
- Platform: Linux-5.10.104-aufs-3-x86_64-with-debian-10.10
- Python version: 3.7.3
- Numpy version: 1.21.6
- PyTorch version (GPU?): 1.11.0+cu113 (True)
- `Accelerate` default config:
        Not found
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

```
from transformers import AutoModelForCausalLM, AutoTokenizer

model = AutoModelForCausalLM.from_pretrained("bigscience/bloom", device_map="auto",  cache_dir="/mnt/local-ssd/hugginface_cache", offload_folder="/mnt/local-ssd/offload_folder", max_memory={"cpu": 17179869184})
```

![image](https://user-images.githubusercontent.com/3121971/174493108-b363a018-725b-4dd4-8bd7-dd9936443761.png)

### Expected behavior

```Shell
This script should Load the Bloom language model, but instead raises exception.
```

**Comment by noamwies** [comment]
Thanks!
To be honest, I have already fixed the issue locally. My main motivation was to run Bloom with the least amount of resources possible, so I did not want to convert into `float32`. As an alternative, I used PyTorch's `tensor.view(torch.int16)` to reinterpret the underlined data as `int16` before converting to numpy, and then after loading the `int16` tensor once again, I reinterpret the underlined data with `tensor.view(torch.bfloat16)`.
@sgugger what did you think on my solution?
